File: spec/functional_test/fixtures/java/javalin_callees/src/Application.java
package com.example;

import io.javalin.Javalin;

public class Application {
    public static void main(String[] args) {
        Javalin app = Javalin.create();

        app.post("/users", ctx -> {
            String name = ctx.queryParam("name");
            String saved = UserService.save(name);
            AuditLog.write(saved);
            ctx.result(saved);
        });

        app.get("/profile", ctx -> {
            String built = buildProfile();
            AuditLog.write(built);
            ctx.result(built);
        });

        app.get("/legacy", ctx -> {
            AuditLog.write("legacy");
            ctx.result(getLegacy().toString());
        });

        app.start();
    }

    static String buildProfile() {
        return "profile";
    }

    static String getLegacy() {
        return "legacy";
    }
}

File: spec/functional_test/fixtures/kotlin/ktor_callees/Application.kt
package com.example.ktor

import io.ktor.server.application.*
import io.ktor.server.request.*
import io.ktor.server.response.*
import io.ktor.server.routing.*

fun Application.configureRouting() {
    routing {
        post("/users") {
            val name = call.parameters["name"]
            val saved = service.save(name)
            AuditLog.write(saved)
            call.respondText(saved)
        }

        get("/profile") {
            val built = this.buildProfile()
            AuditLog.write(built)
            call.respondText(built)
        }

        get("/legacy") {
            AuditLog.write("legacy")
            call.respondText(getLegacy().toString())
        }

        route("/admin") {
            get("/dashboard") {
                renderDashboard()
            }
        }
    }
}
